plugins/module_utils/network/sonic/config/l3_interfaces/l3_interfaces.py

```diff
@@ -143,19 +143,21 @@ def _state_replaced(self, want, have, diff):
         """
         ret_requests = list()
         commands = list()
-        l3_interfaces_to_delete = get_diff(have, want, TEST_KEYS)
-        obj = self.get_object(l3_interfaces_to_delete, want)
-        diff = get_diff(obj, want, TEST_KEYS)
+        new_want = self.update_object(want)
+        get_replace_interfaces_list = self.get_interface_object_for_replaced(have, want)
+
```

Comment:
The deletion portion of this block has the following problems that need to be fixed (in the section from here through the current line 154):

- No deletion requests should be sent to the device for interfaces that currently have no L3 configuration. The test output indicates that a "delete" request is sent to interface Etherneret1/6 for a "replaced" configuration that touches that interface even though the "before" L3 configuration for this interface is empty.
- In the reported output for "deleted" commands, the deleted values displayed should be the values that were configured at the start of execution of the task (the values that were actually deleted). This corresponds to the subset of "have" specified in "get_replace_interfaces_list", not to "want" as currently coded in line 154.

Reply:
After the most recent push, the revised behavior looks fine. 

Thanks for making this additional revision.

Resolved.